### ISSUE TITLE
More flexibility in file type-assembly

### ISSUE THREAD

**Issue by ARWattam** [PATRIC3/patric3_website#1991]
A user reports: 
I have sent several reports since I participated in the PATRIC workshop 
last week, because my assemblies on own data continued to fail. I just 
wanted to let you know that I have found the problem. It seems the read 
files must have the extension .fastq.gz or at least not be named like mine 
were initially (e.g., SRR4114375_2.fastq.cut.trim.gz.headers.fixed.gz). 
When I changed the filenames to, e.g., 
SRR4114375_2.cut.trim.headers.fixed.fastq.gz the SPAdes assembly succeeded.

**Comment by mshukla1** [comment]
I think the general issue is, if there are file extensions not supported by the underlying services, we should prevent them from being uploaded in the first place. 

For example, one of the user assembly jobs failed with the following error. 

== Error ==  file with reads has unsupported format (only .fq, .fastq, .bam, .fa, .fasta, .fq.gz, .fastq.gz, .bam.gz, .fa.gz, .fasta.gz are supported): /disks/arast/data/zmerchant_patricbrc_org/32/raw/trimmed_UMB0906_R2.fastq.txt (right reads, library number: 1, library type: paired-end)

Assembly jobs worked after changing file names from fastq.txt to fastq.

**Comment by dawenx** [comment]
I will look into this.

I need some help though. In particular, @hyoo, for a given WorkspaceObjectSelector type, where can I find documentation on its supported format, e.g.,  ".fq, .fastq, .bam, .fa, .fasta, .fq.gz, .fastq.gz, .bam.gz, .fa.gz, .fasta.gz" for "reads" in this case? Thanks!

**Comment by hyoo** [comment]
lets start app defines proper input and its file type.

1. reads: assembly, varation, rna-seq, tn-seq, metagenomics
   => file name has to be one of .fq, .fastq, .bam, .fa, .fasta, .fq.gz, .fastq.gz, .bam.gz, .fa.gz, .fasta.gz 
2. contigs: annotation.
   => file name has to be ?

**Comment by dawenx** [comment]
Per our discussion in the science meeting on 08/02/2008, the fix I have is better than what we have on the production, so I checked in my changes and pushed them to my dev branch. This would address all services/type on the front end UI assuming user does not disable it by un-checking the restriction checkbox. @hyoo do I need to submit a PR in this case?

There's a separate/related issue with the drag and drop (dnd) and I have communicated this with @nconrad and he will look into it. Neal, the change is in my dev branch, i.e., https://github.com/dawenx/p3_web/commit/0782b9b4bad58a5b1392906736afb8f3915f6896
